### DISCUSSION TITLE
How to  reduce the resource usage of CPU and GPU

### DISCUSSION THREAD

**Discussion by dugufly** [google/filament#4453] [Q&A]
I am running this program on  vivo X50 (Android, the chip is SM7250, the refresh rate is 60FPS):  hellotriangle

CPU usage: 42% GPU usage: 59%

Is there any way to reduce the resource usage of CPU and GPU?

**Comment by romainguy** [comment]
The % of utilization may not be relevant without knowing the current frequency of the CPU and GPU. The systrace/perfetto tools will give you that information. 42% of CPU usage at the lowest clock would definitely not be an issue. You can easily reduce GPU usage by lowering the rendering resolution.

**Reply by pixelflinger** [reply]
Correct, for hello-triangle the GPU is usually clocked at 250MHz and the various buses at low frequencies as well.
